test/YieldDistributor.t.sol

```diff
@@ -446,3 +450,295 @@ contract YieldDistributorTest is Test {
+
+        // Create an instance of the deployed contract
+        nftMultiplier = NFTMultiplier(deployedAddress);
+
+        // Verify the deployment
+        assertEq(address(nftMultiplier.NFTAddress()), nftContractAddress);
+        assertEq(nftMultiplier.multiplyingFactor(), multiplyingFactor);
+        assertEq(nftMultiplier.validUntil(address(this)), validUntilBlock);
+
+        // Test integration with YieldDistributor
+
+        // Set up a voter with an NFT
+        address voter = address(0x1);
+        vm.mockCall(nftContractAddress, abi.encodeWithSelector(IERC721.balanceOf.selector, voter), abi.encode(1));
+
+        // Set up voting
+        address[] memory voters = new address[](1);
+        voters[0] = voter;
+        setUpAccountsForVoting(voters);
+        setUpForCycle(yieldDistributor);
+        uint256 initialVotingPower = yieldDistributor.getCurrentVotingPower(voter);
+
+        yieldDistributor.addMultiplier(IMultiplier(address(nftMultiplier)));
+
+        uint256[] memory points = new uint256[](1);
+        points[0] = 100;
+
+        // Cast vote
+        vm.prank(voter);
+        yieldDistributor.castVote(points);
+
+        // Check that the voting power was multiplied correctly
+        uint256 expectedVotingPower = (initialVotingPower * multiplyingFactor) / 1e18;
+        assertEq(yieldDistributor.projectDistributions(0), expectedVotingPower);
+    }
+
+    function setUpVoteWithVariableMultipliers(uint256 numMultipliers) internal {
+        // Set up a voter
+        address voter = address(0x1);
+        address[] memory voters = new address[](1);
+        voters[0] = voter;
+
+        setUpAccountsForVoting(voters);
+        setUpForCycle(yieldDistributor);
+        uint256 initialVotingPower = yieldDistributor.getCurrentVotingPower(voter);
+        uint256 seed = block.timestamp;
+        MockMultiplier[] memory mockMultipliers = new MockMultiplier[](numMultipliers);
+        uint256 validUntil = type(uint256).max;
+        uint256 totalFactor = 0;
+        for (uint256 i = 0; i < numMultipliers; i++) {
+            uint256 factor = 1e18 + (uint256(keccak256(abi.encode(seed, i))) % 1e18);
+            totalFactor += factor;
+            mockMultipliers[i] = new MockMultiplier();
+            mockMultipliers[i].setMultiplier(factor, validUntil);
+            yieldDistributor.addMultiplier(mockMultipliers[i]);
+        }
+
+        uint256[] memory points = new uint256[](1);
+        points[0] = 100;
+
+        vm.startPrank(voter);
+        yieldDistributor.castVote(points);
+
+        // Calculate expected voting power
+        uint256 expectedVotingPower = (initialVotingPower * (totalFactor)) / yieldDistributor.PRECISION();
+
+        // Check that the voting power was multiplied correctly
+        assertEq(yieldDistributor.projectDistributions(0), expectedVotingPower);
+        vm.stopPrank();
+    }
+
+    function testCastVote100Multipliers() public {
+        setUpVoteWithVariableMultipliers(100);
```

Comment:
nothing is asserted in these lasts tests

Reply:
addressed with f867e06